### DISCUSSION TITLE
Migrating an existing Qhub to Version 0.3?

### DISCUSSION THREAD

**Discussion by rsignell-usgs** [nebari-dev/nebari#491] [Q&A] [answered]
@jkellndorfer and I would of course like to upgrade to the fabulous features of Qhub 0.3! 
I imagine other existing Qhub users would be interested in this as well. 

What procedure do you recommend we follow?

Also: @laisbsc , is this the best place to discuss?

**Comment by dharhas** [comment]
@costrouc @tylerpotts Lets get the outline of what to do in this issue and also find a time to be available when Rich and Josef actually kick off the process...

**Reply by rsignell-usgs** [reply]
Are we ready to schedule this or should we wait for a more stable release?

**Comment by brl0** [comment]
This is not an answer, but I wanted to share some previous experiences for information sharing purposes... these are based on older versions before many recent changes, and may or may not have been appropriate to begin with, so caution is warranted when considering trying any of these things.

A few weeks ago, when I was testing the upgrade path for dask-gateway, I found that first uninstalling the helm chart for dask-gateway before running the deploy was necessary.  That was done with a command like this: `helm uninstall dask-gateway --namespace dev`.  I have not tested with more recent versions, and this may have changed, especially since @costrouc has moved away from using the dask-gateway helm chart at all, but it wouldn't surprise me if this is needed when upgrading an existing qhub.

I was also testing deployment of the latest version of jupyterhub, so in that case I also uninstalled that helm chart before deployment.

Basically, in my very limited experience, terraform did not seem to properly handle redeployment of certain helm resources, I presume because terraform does not directly manage the state of those resources, which I suppose is a potential upshot of qhub moving away from using helm charts.

The old nginx ingress controller, which was deployed with helm as `ingress`, is now being replaced with traefik, so I wonder if the old ingress helm chart will need to be uninstalled manually as well.

This might be helpful for troubleshooting possible issues while upgrading, but please don't blow up any existing installations based on this without first consulting with the qhub team.

**Comment by tylerpotts** [comment, marked as answer]
For future readers, the solution we went with was backing up the data and restoring to a new 0.3 deployment using the process [described here](https://docs.qhub.dev/en/stable/source/04_how_to_guides/4_system_backup.html?highlight=backup) and then manually deleting the AWS resources via the console with the steps [described here](https://github.com/Quansight/qhub/discussions/669?converting=1)

**Comment by Adam-D-Lewis** [comment]
I'm closing old discussions in a repo cleanup effort.  Feel free to re-open if needed.
